main.tf

```diff
@@ -134,7 +134,7 @@ module "lambda" {
 
   allowed_triggers = {
     AllowExecutionFromSNS = {
-      principal  = "sns.amazonaws.com"
+      principal  = "sns.${data.aws_partition.current.dns_suffix}"
```

Comment:
is this warranted?

Reply:
This is going to allow the module to be used for other partitions like china or gov cloud.
It doesn't harm :)

Reply:
Are you saying it doesn't work in those partitions? we haven't had reports of this (nor requests) - and some service principals are not partition specific. we shouldn't change this unless we know for certain that its correct and required by users

Reply:
Yup good catch, my bad. rolled that back.
source for CN: https://docs.amazonaws.cn/en_us/sns/latest/dg/sns-access-policy-use-cases.html